_posts/2015-12-13-testing-multithreaded-code-in-java.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+name: Testing-multithreaded-code-in-java
+title: 'Testing multithreaded code in Java'
+date: 2015-12-13 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/picot.png
+tags:
+- testing
+- concurrency
+- java
+
+---
+
+Testing multithreaded code is a tough challenge. The first advice that you get when try to test concurrency is to isolate your concurrent concerns in the code as much as possible. This a general design advice but in this case it's even more important. When you have properly unit tested the logic that is wrapped by the concurrent construct, you can start thinking about how to test that bit. Otherwise you can spend so many time trying to figure out what's the concurrent problem, being in the end a simple flawed business logic.
```

Comment:
Otherwise you might spend a long time trying to figure out a concurrency problem that turns out to be flawed business logic in the end.